inspect.cpp

```diff
@@ -416,43 +416,91 @@ namespace Sass {
 
   void Inspect::operator()(Number* n)
   {
+
+    string res;
+
+    // init stuff
     n->normalize();
+    int precision = 5;
+    double value = n->value();
+    // std::num_put<char> facet
+    // get rid of negative zero
+    // if (value == 0) value *= value;
+    // get option from optional context
+    if (ctx) precision = ctx->precision;
+
+    // check if the fractional part of the value equals to zero
+    // neat trick from http://stackoverflow.com/a/1521682/1550314
+    // double int_part; bool is_int = modf(value, &int_part) == 0.0;
+
+    // first sample
     stringstream ss;
-    ss.precision(ctx ? ctx->precision : 5);
-    ss << fixed << n->value();
-    string d(ss.str());
-    // store if the value did not equal zero
-    // if after applying precsision, the value gets
-    // truncated to zero, sass emits 0.0 instead of 0
-    bool nonzero = n->value() != 0;
-    size_t decimal = d.find('.');
-    if (decimal != string::npos) {
-      for (size_t i = d.length()-1; d[i] == '0' && i >= decimal; --i) {
-        d.resize(d.length()-1);
+    ss.precision(12);
+    ss << value;
+
+    // check if we got scientific notation in result
+    if (ss.str().find_first_of("e") != string::npos) {
+      ss.clear(); ss.str(string());
+      ss.precision(max(12, precision));
+      ss << fixed << value;
+    }
+
+    const char *it = ss.str().c_str();
+    while (*it && *it != '.' && *it != ',') ++ it;
```

Comment:
Unoftunately every bit (even the number 12) seems to be exactly needed this way.  But I guess this one loop can be rewritten with find_first_of ...